cmd/neofs-node/container.go

```diff
@@ -351,6 +354,82 @@ func setContainerNotificationParser(c *cfg, sTyp string, p event.NotificationPar
 	c.cfgContainer.parsers[typ] = p
 }
 
+func initPaymentChecker(c *cfg) {
+	var (
+		l = c.log.With(zap.String("component", "paymentChecker"))
+		p = &paymentChecker{
+			m:          sync.RWMutex{},
+			statuses:   make(map[cid.ID]int64),
+			balanceCli: c.bCli,
+		}
+	)
+
+	const changeUnpaidStatusEventName = "ChangePaymentStatus"
+	typ := event.TypeFromString(changeUnpaidStatusEventName)
+	c.cfgBalance.parsers[typ] = balanceEvent.ParseChangePaymentStatus
+	c.cfgBalance.subscribers[typ] = append(c.cfgBalance.subscribers[typ], event.WorkerPoolHandler(c.cfgBalance.workerPool, func(e event.Event) {
+		ev := e.(balanceEvent.ChangePaymentStatus)
+		p.m.Lock()
+		defer p.m.Unlock()
+
+		cID := cid.ID(ev.ContainerID)
+		if ev.Unpaid {
+			l.Info("container status has changed to unpaid",
+				zap.Stringer("cID", cID),
+				zap.Uint64("epoch", ev.Epoch))
+
+			p.statuses[cID] = int64(ev.Epoch)
+		} else {
+			l.Debug("container status has changed to paid",
```

Comment:
Info?

Consider the following scenario as well:
 * container gets "unpaid" status
 * all data is deleted from nodes
 * its size is zero, so there are no settlements to make
 * but its status is still "unpaid" even if owner has some balance

Can this be a problem?